### ISSUE TITLE
Middle, numerous whitespaces, falsely pass MD039

### ISSUE THREAD

**Issue by psyomn** [markdownlint/markdownlint#225]
https://github.com/markdownlint/markdownlint/issues/182#issuecomment-385262804

* * *

Another issue: multiple spaces in the middle of link text passed checks.

```bash
echo '[foo     bar](http://example.org) {MD039}' | mdl && echo "passed"
passed
```

* * *
Also check: 

> Confirmed with [hello "world")](https://purpleidea.com/)
https://github.com/markdownlint/markdownlint/issues/182#issuecomment-367546262

**Comment by DavidAnson** [comment]
MD039 is only documented to detect spaces surrounding the link text:
https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md039---spaces-inside-link-text

**Comment by psyomn** [comment]
Thanks! Fair point, but I think this might be a good addition in the future. Would you be ok with the documentation to be updated, or would you rather have this as some other rule?

**Comment by DavidAnson** [comment]
No strong feeling, just defending the current rule. :) I wonder if “no multiple spaces” makes sense as a more general rule that would apply here, in emphasis, headers, etc.?

**Comment by psyomn** [comment]
I think that sounds pretty good actually. This can also check if you're not doing funny things inside paragraphs:

```
You have a lot of text, with a lot of text, but oh noees    extraneous space.
```

Perhaps another addition would be to support double spacing after periods in this rule as well. 

```
You have a sentence.  And notice the double spaces there. 
```

I've seen some people, and some manuals do this, so I think it might be appreciated. I'm wondering whether md039 should implement this, and include another rule for the rest of possibilities, or just create a new rule, that covers everything concerning extraneous whitespace. Overall I'd like not to break current usage of `mdl` on other projects, as I'd like to focus more on stabilizing and fixing bugs for now.

**Comment by DavidAnson** [comment]
I like doing this in a new rule to avoid changing the meaning of the current one. Now that I check, here is the same idea and some related ones from the Node.js project: https://github.com/DavidAnson/markdownlint/issues/51
